R/CoNGA.R

```diff
@@ -0,0 +1,120 @@
+  file <- files[files == 'run_CoNGA.py']
+  str <- readr::read_file(file)
+  script <- tempfile()
+  readr::write_file(str, script)
+  newstr <- paste0("run_CoNGA(features_file = '", features_file,
+                   "',tcr_datafile = '", tcr_datafile, "', gex_datafile = '",
+                   gex_datafile, "', organism = '", organism,
+                   "', outfile_prefix = '", outfile_prefix,"', gex_datatype = '",
+                   gex_datatype, "', clones_file = '", clones_file,
+                   "', outfile_prefix_for_qc_plots = '", 
+                   outfile_prefix_for_qc_plots,"')")
+  readr::write_file(newstr, script, append = TRUE)
+  system2(reticulate::py_exe(), script)
+}
+
+#' @title Run CoNGA on Seurat data and subsets
+#'
+#' @description Runs CoNGA on combo data and on combo data split apart into individual samples
+#' @param seuratObj input Seurat object
+#' @param splitFields a vector of metadata columns used for splitting the Seurat object
+#' @param outputdir file path of directory for output files
+#' @param organism 'human' or 'rhesus'
+#' @param gex_datatype "10x_h5"
+#'
+#' @export
+runCoNGA_on_seuratdata_and_subsets <-function(seuratObj, splitFields,
+                                              outputdir, organism, gex_datatype) {
+  dir.create(outputdir)
+  maindir <- paste0(outputdir, "/main")
+  dir.create(maindir)
+  SeuratToCoNGA(seuratObj, maindir)
+  clonesfile <- paste0(maindir, "/TCRs.csv")
+  clones <- read.csv(clonesfile)
+  meta <- seuratObj@meta.data
+  clones_sub <- subset(clones, subset = barcode %in% rownames(meta))
+  write.csv(clones_sub, paste0(maindir, "/TCRs_sub.csv"), row.names = F)
+  grouplist <- list()
+  for (i in 1:length(splitFields)) {
+    grouplist[[i]] <- meta[,splitFields[i]]
+  }
+  splitlist <- split(meta, grouplist)
+  barcodelist <- lapply(splitlist, rownames)
+  
+  features_file = paste0(maindir, "/varfeats.csv")
+  outfile_prefix = paste0(maindir, "/")
+  clones_file = paste0(maindir, "/clones.tsv")
+  tcr_datafile <- paste0(maindir, "/TCRs_sub.csv")
+  gex_datafile <- paste0(maindir, "/GEX.h5")
+  outfile_prefix_for_qc_plots= paste0(maindir, "/QC_plots/")
+  RunCoNGA(features_file, tcr_datafile, gex_datafile, organism,
+           outfile_prefix, gex_datatype, clones_file,
+           outfile_prefix_for_qc_plots)
+  
+  for (splitname in names(barcodelist)) {
+    if (length(barcodelist[[splitname]]) > 0){
+      current_dir <- paste0(outputdir, "/", splitname)
+      dir.create(current_dir)
+      clones_sub <- subset(clones, subset = barcode %in% barcodelist[[splitname]])
+      tcr_datafile <- paste0(current_dir, "/TCRs_sub.csv")
+      write.csv(clones_sub, tcr_datafile, row.names = F)
+      gex_datafile <- paste0(current_dir, "/GEX.h5")
+      seuratObj_sub <- subset(seuratObj, cells = barcodelist[[splitname]])
+      seuratObj_sub <- NormalizeAndScale(seuratObj_sub)
+      DropletUtils::write10xCounts(x = seuratObj_sub@assays$RNA@counts, path = gex_datafile)
```

Comment:
exposing the the assay as an argument and setting the default argument as RNA is probably desirable.